nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/UnpackContent.java

```diff
@@ -139,6 +145,20 @@ public class UnpackContent extends AbstractProcessor {
                     PackageFormat.FLOWFILE_STREAM_FORMAT_V2.toString(), PackageFormat.FLOWFILE_TAR_FORMAT.toString())
             .defaultValue(PackageFormat.AUTO_DETECT_FORMAT.toString())
             .build();
+    public static final PropertyDescriptor ZIP_FILENAMES_ENCODING = new PropertyDescriptor.Builder()
+            .name("zip-filenames-encoding")
+            .displayName("Zip Filenames Encoding")
+            .description(
+                "The encoding used by zip creating utility, for the file names inside the zip. Processor will pass this encoding to Zip unpacker. For example 'Cp437', 'UTF8' etc. Default is to "
```

Comment:
Consider alternative text

'If supplied this character set will be supplied to the Zip utility to attempt to decode filenames using the specific character set.  If not specified the default platform character set will be used.  This is useful if a Zip was created with a different character set than the platform default and the zip uses non standard values to specify.

Reply:
Instead of describing the Cp437 scenario in the property description consider creating an '@UseCase' annotation to describe the specific case and how the whole processor would be configured.  See 'PublishSlack' for an example of this done well.